File: src/main/python/com.hazza.LeetCodeByPython/SegmentOne/No1_Two_Sum.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Created by HazzaCheng on 2019/4/1 19:30
# https://leetcode.com/problems/two-sum/
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i, num in enumerate(nums):
            if target - num in map:
                return [map[target - num], i]
            map[num] = i

        return []


if __name__ == '__main__':
    solution = Solution()
    nums = [2, 7, 11, 15]
    assert solution.twoSum(nums, 9) == [0, 1]

File: src/main/java/com/hazza/leetcode/SegmentOne/No1TwoSum.java
package com.hazza.leetcode.SegmentOne;

import java.util.HashMap;
import java.util.Map;

/**
 * https://leetcode.com/problems/two-sum/#/description
 */
public class No1TwoSum {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i = 0;i < nums.length; ++i) {
            if (map.containsKey(target - nums[i])) {
                result[1] = i;
                result[0] = map.get(target - nums[i]);
                return result;
            }
            map.put(nums[i], i);
        }
        return result;
    }
}
